nixos/modules/programs/corefreq.nix

```diff
@@ -0,0 +1,38 @@
+
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.programs.corefreq;
+  kernelPackages = config.boot.kernelPackages;
+  corefreq = cfg.package;
+in {
+  options = {
+    programs.corefreq = {
+      enable = lib.mkEnableOption "Whether to enable the corefreq daemon and kernel module";
+
+      package = lib.mkOption {
+        type = lib.types.package;
+        default = kernelPackages.corefreq;
+        defaultText = lib.literalExpression "config.boot.kernelPackages.corefreq";
+        description = ''
+          The corefreq package to use.
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ corefreq ];
+    boot.extraModulePackages = [ corefreq ];
+
+    # Create a systemd service for the corefreq daemon
+    systemd.services.corefreq = {
+      description = "CoreFreq daemon";
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStartPre = "${pkgs.kmod}/bin/modprobe corefreqk";
```

Comment:
kmod should be configurable too

Reply:
I'm looking for an example module that configures kmod so I have an idea of the naming convention to use, but I can't find any. 

Would you just add a `kmod` option with `types.package`?

Reply:
Yes, it should suffice.

Reply:
Done.